lib/livebook/teams/requests.ex

```diff
@@ -216,6 +218,27 @@ defmodule Livebook.Teams.Requests do
     delete("/api/v1/org/deployment-groups/agent-keys", params, team)
   end
 
+  @doc """
+  Send a request to Livebook Team API to deploy an app.
+  """
+  @spec deploy_app(Team.t(), AppDeployment.t()) ::
+          {:ok, map()} | {:error, map() | String.t()} | {:transport_error, String.t()}
+  def deploy_app(team, app_deployment) do
+    secret_key = Teams.derive_key(team.teams_key)
+
+    params = %{
+      filename: app_deployment.filename,
```

Comment:
Since we're encrypting the content, can we adjust the filename accordingly?

Req does content-type AND filename sniffing. We discussed that `Req.get!(archive_url)` was crashing when the `content-type: application/zip` was sent but the content is not actually zip. It will crash if the filename is `.zip` too/

```suggestion
      filename: app_deployment.filename <> ".encrypted",
```